server/modules/taxonomy/repository/taxonomyRepository.js

```diff
@@ -88,10 +88,14 @@ export const fetchTaxonomyByUuid = async (surveyId, uuid, draft = false, client
     (record) => dbTransformCallback(record, draft, true)
   )
 
-export const fetchTaxonomiesBySurveyId = async ({ surveyId, draft = false, limit = null, offset = 0 }, client = db) =>
+export const fetchTaxonomiesBySurveyId = async (
+  { surveyId, draft = false, limit = null, offset = 0, search = '' },
+  client = db
+) =>
   await client.map(
     `SELECT * 
-     FROM ${getSurveyDBSchema(surveyId)}.taxonomy 
+     FROM ${getSurveyDBSchema(surveyId)}.taxonomy
+     ${search ? `WHERE ${DbUtils.getPropColCombined(Taxonomy.keysProps.name, draft)} ILIKE '%${search}%'` : ''} 
```

Comment:
also in description props field.

Reply:
sure

Reply:
makes sense extract this to some utils function?

``` ${search ? `WHERE 
      ${DbUtils.getPropColCombined(Taxonomy.keysProps.name, draft)} ILIKE '%${search}%'
      OR 
      ${DbUtils.getPropColCombined(Taxonomy.keysProps.descriptions, draft)} ILIKE '%${search}%'
      ` : ''} ```

``` 
searchByFields({ fields, search }) => fields.reduce((q,f,i, arr) => `${q}${f} ILIKE '%${search}%' ${i < arr.length ? 'OR' : ''}` ,'')

${search ? `WHERE ${DbUtils.searchByFields({fields: [
DbUtils.getPropColCombined(Taxonomy.keysProps.name, draft),
DbUtils.getPropColCombined(Taxonomy.keysProps.descriptions, draft)], search })` : ''} ```